### ISSUE TITLE
please help on tumbler.py problem

### ISSUE THREAD

**Issue by kuje11** [JoinMarket-Org/joinmarket#739]
hello,
I got that problem on irc i think when i try to tumbling my wallet to 5 btc addresses ,

connecting to host irc.cyberguerrilla.org6697
failed to send ping message on socket
disconnected from irc host irc.cyberguerrilla.org6697
connecting to host irc.cyberguerrilla.org6697
disconnected from irc host irc.cyberguerrilla.org6697
irc ping timed out
errored while trying to quit: error(10057, 'Une requ\xeate d\x92envoi ou de r\xe9ception de donn\xe9es n\x92a pas \xe9t\xe9 autoris\xe9e car le socket n\x92est pas connect\xe9 et (lors de l\x92envoi sur un socket datagramme en utilisant un appel sendto) aucune adresse n\x92a \xe9t\xe9 fournie')

**Comment by goodpunk6** [comment]
can confirm. I have the same exact issue

**Comment by AdamISZ** [comment]
This is a problem connecting to irc.cyberguerrilla.org on port 6697. Things to know:

* There are two connections configured by default (the above, and agora.anarplex.net). If *during* running, one of them goes down, the other should continue as normal (although you may get a fair amount of network warning messages). If you fail to connect to either one *at start up* of script, the script will not start up at all.

* In the latter case, you would need to edit the joinmarket.cfg file to remove the IRC server that was refusing connections (each of the settings in MESSAGING is a comma separated list, by default with 2 items each, as above, you would have to edit to have one entry in each of those settings.

Hope it helps.

**Comment by goodpunk6** [comment]
Are there other servers you recommend, that I can add or does it have to be these two? Cuz I'm technically only using one since cyberguerrilla won't connect. Its odd that this is happening to more than one person.

**Comment by AdamISZ** [comment]
Those are the only two in general usage, as far as I know. It's difficult to find public IRC servers that support Tor (which a lot of users prefer to use for sensible reasons). If a group of users want to use other servers, they can, but obviously the network effect kind of dictates we all use the same ones.

It's not odd that it happens to more than one person; if the IRC server is having problems, it'd be normal for either many or all users to experience the connection problem.

**Comment by goodpunk6** [comment]
I'm just trying to rule out that joinmarket is not working. I guess their
server is just having issues. I'm not too happy that this isn't working
right now and that a fix seems unobtainable. I can't send money so
joinmarket has cost me $160 so far with no solution in sight. This sucks.

**Comment by AdamISZ** [comment]
Sorry, it's a little unclear what you mean; have you tried setting the server to agora.anarplex.net only? (removing the entries for cyberguerrilla). Has the tumbler started but not completed (I presume so, from what you say)? As to an amount it's cost you, are you referring to fees or to funds that are "in transit", or have gone missing? One general point that may be helpful whatever your situation is: note that you can send funds out of the wallet directly, without using IRC and coinjoin, using `python sendpayment.py -N 0 ...` (see `python sendpayment.py --help` for some info.)

**Comment by goodpunk6** [comment]
I have not tried setting it agora alone. Does that make a difference?
Meaning, will I be able to connect to the server, tumble my coins and then
send them?

The tumbler does not tumble anything even when I use 3 addresses. I get the
same exact errors.

I created a separate issue for sending money -
https://github.com/JoinMarket-Org/joinmarket/issues/740 - I have the same
exact issue when I try to send money. It simply won't. I was able to send
my money to the joinmarket wallet that I created with joinmarket. But I
can't send the money out from joinmarket to where i want it to be, so the
money has basically gone poof. Since I can't send it out or tumble it, its
basically lost right now.

I tried - python sendpayment.py -N 0 wallet 0 wallet address - but got some
utxos error.

If I try to run - python sendpayment.py wallet amount address - no matter
what amount I use, i get a not enough funds error. Despite that the wallet
has $160 in it. Any amount.

**Comment by goodpunk6** [comment]
Ok, so removing cyberguerrilla seems to have made things a ton better. I am happy that I was prompted y/n for a bunch of orders.
